### PR TITLE
#182201714 Setup docker for backend application 

### PR DESCRIPTION
**Pull request by uwambajeddy**
#### What does this PR do?
This pull request contains initial setup of  docker for backend application 

#### How should this be manually tested?
- After cloning this PR you will cd into it. you are asked to add environment variables in a file you create on your own called `.env` , follow the instructions written in a file called `.env.example` and Run `docker compose up` . the project will start with a message `Listening on port YOUR_PORT`
- You can  open postman to test users end point by using this URL  `/users` on a `GET` request

#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/182201714

### CODE REVIEW

**Review by semy-tuyi — DISMISSED** [review]

**Review by joshwambere — COMMENTED** [review]
@uwambajeddy  is it a good approach to run  migrations and seed  every time you run the application

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+            - PGPORT=5432
+            - DEV_PGDATABASE=${DEV_PGDATABASE}
+            - TEST_PGDATABASE=${TEST_PGDATABASE}
+            - PROD_PGDATABASE=${PROD_PGDATABASE}
+            - PGUSER=${PGUSER}
+            - PGPASSWORD=${PGPASSWORD}
+            - PGHOST=node_db
+        depends_on: 
+            - node_db
+    migration:
```
Comment (joshwambere) [inline docker-compose.yml:27]: @uwambajeddy  is it a good approach to run migrations and seeds every time the app starts?

Comment (uwambajeddy) [reply docker-compose.yml:27]: Copy that, thanks for the feedback.

**Review by alaintresor — DISMISSED** [review]

**Review by mwisemarierose — DISMISSED** [review]

**Review by Placide123 — DISMISSED** [review]

File: package.json [hunk 2]
```diff
@@ -3,9 +3,32 @@
   "version": "1.0.0",
   "description": "Make company global travel and accommodation easy and convenient for the strong workforce of savvy members of staff, by leveraging the modern web.",
   "main": "src/index.js",
+  "type": "module",
   "scripts": {
-    "start": "node ./src/index.js",
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "start": "NODE_ENV=production node --exec babel-node ./src/index.js",
```
Comment (alaintresor) [inline package.json]: please remember to put cross-env before NODE_ENV

**Review by alaintresor — CHANGES_REQUESTED** [review]
please in package.json

File: package.json (hunk 2 above)
Comment (uwambajeddy) [reply package.json]: copy that!

File: package.json [hunk 3]
```diff
@@ -35,22 +52,27 @@
...
     "cors": "^2.8.5",
     "cross-env": "^7.0.3",
     "dotenv": "^16.0.1",
     "express": "^4.18.1",
+    "joi": "^17.6.0",
+    "jsonwebtoken": "^8.5.1",
     "pg": "^8.7.3",
     "pg-hstore": "^2.3.4",
-    "sequelize": "^5.21.6",
-    "swagger-ui-express": "^4.4.0"
```
Comment (joshwambere) [inline package.json]: why did you remove this dependency

**Review by joshwambere — CHANGES_REQUESTED** [review]

File: package.json (hunk 3 above)
Comment (joshwambere) [reply package.json]: and also docker app is not working as expected

Screenshot from 2022-05-29 20-33-48

Comment (joshwambere) [reply package.json]: same issue applies to this [PR](https://github.com/atlp-rwanda/bravo-bn-be/pull/11)

**Review by Placide123 — APPROVED** [review]
